protocol/x/subaccounts/keeper/transfer.go

```diff
@@ -219,70 +218,61 @@ func (k Keeper) WithdrawFundsFromSubaccountToAccount(
 func (k Keeper) DistributeFees(
 	ctx sdk.Context,
 	assetId uint32,
-	quantums *big.Int,
-	perpetualId uint32,
+	revShares []revsharetypes.RevShare,
+	fill clobtypes.FillForProcess,
 ) error {
 	// get perpetual
-	perpetual, err := k.perpetualsKeeper.GetPerpetual(ctx, perpetualId)
+	totalFeeQuoteQuantums := new(big.Int).Add(fill.TakerFeeQuoteQuantums, fill.MakerFeeQuoteQuantums)
+	perpetual, err := k.perpetualsKeeper.GetPerpetual(ctx, fill.ProductId)
 	if err != nil {
 		return err
 	}
 
-	collateralPoolAddr, err := k.GetCollateralPoolFromPerpetualId(ctx, perpetualId)
+	collateralPoolAddr, err := k.GetCollateralPoolFromPerpetualId(ctx, fill.ProductId)
 	if err != nil {
 		return err
 	}
 
 	// calculate market mapper rev share
-	marketMapperShare := big.NewInt(0)
-	revShareAddr, revSharePpm, err := k.revShareKeeper.GetMarketMapperRevenueShareForMarket(
-		ctx,
-		perpetual.Params.MarketId,
-	)
-	// Note: The likelihood of this error is very low, and not getting the rev share should not
-	// prevent the trade from going through. Therefore, we log the error and continue
-	if err != nil {
-		log.ErrorLog(ctx, "DistributeFees: failed to get market mapper revenue share", "err", err)
-	}
-	if err == nil && revShareAddr != nil {
-		if revSharePpm >= 1e6 {
-			log.ErrorLog(
-				ctx,
-				"DistributeFees: revSharePpm is greater than or equal to 100%",
-				"revSharePpm",
-				revSharePpm,
-			)
-		} else {
-			// marketMapperShare = quantums * revSharePpm / 1e6
-			marketMapperShare.Div(
-				new(big.Int).Mul(quantums, big.NewInt(int64(revSharePpm))),
-				big.NewInt(1e6),
-			)
-		}
+	totalRevShareQuoteQuantums := big.NewInt(0)
+
+	for _, revShare := range revShares {
+		totalRevShareQuoteQuantums.Add(totalRevShareQuoteQuantums, revShare.QuoteQuantums)
```

Comment:
This can be done in the for loop [below](https://github.com/dydxprotocol/v4-chain/pull/2251/files#diff-9b14a21a3effc6a901dba396d40a3fc43d7459350af7c9c8da040cff3606c623R262)

Reply:
Note all these logic is processed atomically - if any thing propagates an error to the top level, then the entire transaction is reverted